### DISCUSSION TITLE
Tree: expand/collapse on entire node click, not only caret

### DISCUSSION THREAD

**Discussion by warpdesign** [palantir/blueprint#5885] [Q&A] [answered]
#### Feature request
Right now, you have to click on the caret to collapse/expand a tree node: I wish there was a prop that would make clicking anywhere in the tree node collapse/expand the node's children.

**Comment by adidahiya** [comment, marked as answer]
How are you constructing your Tree? Can you share some code? You get the callbacks to decide when nodes are expanded and selected. In the [docs example](https://github.com/palantir/blueprint/blob/2eddfdea0a55ccef89b1cccc43a352abcda52477/packages/docs-app/src/examples/core-examples/treeExample.tsx#L45), we choose to use `onNodeClick` to select nodes, but you should be able to configure it to expand/collapse nodes instead.
